docs/blog/things-to-know-when-developing-with-lit-element/index.md

```diff
@@ -0,0 +1,233 @@
+  public constructor() {
+    super();
+    this.addEventListener('known-event-name', this.eventHandler);
+  }
+
+  private eventHandler(): void {
+    // do stuff.
+  }
+}
+```
+
+Done and done!
+
+## styleMap usage
+
+Contrary to what you might read in the documentation site (PRs are incoming to correct, so you may miss it altogether!), `styleMap()` does accept `[name: string]: string | undefined | null;` which allows you to leverage `undefined` to prevent a property from being added to the style of an element, e.g.:
+
+```js
+import {styleMap} from 'lit/directives/style-map.js';
+// ...
+
+  render() {
+    return html`
+      <p
+        style=${styleMap({
+          border: '1px solid',
+          width: '200px',
+          float: undefined, // I don't show a type mismatch
+          margin: 10, // I do show a type mismatch
+        })}
+      >Hello, world!</p>`;
+  }
+```
+
+[Check it out here.](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCB7c3R5bGVNYXB9IGZyb20gJ2xpdC9kaXJlY3RpdmVzL3N0eWxlLW1hcC5qcyc7XG5cbkBjdXN0b21FbGVtZW50KCdzaW1wbGUtZ3JlZXRpbmcnKVxuZXhwb3J0IGNsYXNzIFNpbXBsZUdyZWV0aW5nIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIHN0YXRpYyBzdHlsZXMgPSBjc3NgcCB7IGNvbG9yOiBibHVlIH1gO1xuXG4gIEBwcm9wZXJ0eSgpXG4gIG5hbWUgPSAnU29tZWJvZHknO1xuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgPHBcbiAgICAgICAgc3R5bGU9JHtzdHlsZU1hcCh7XG4gICAgICAgICAgYm9yZGVyOiAnMXB4IHNvbGlkJyxcbiAgICAgICAgICB3aWR0aDogJzIwMHB4JyxcbiAgICAgICAgICBmbG9hdDogdW5kZWZpbmVkLFxuICAgICAgICAgIG1hcmdpbjogMTAsXG4gICAgICAgIH0pfVxuICAgICAgICA-SGVsbG8sICR7dGhpcy5uYW1lfSE8L3A-YDtcbiAgfVxufVxuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxzaW1wbGUtZ3JlZXRpbmcgbmFtZT1cIldvcmxkXCI-PC9zaW1wbGUtZ3JlZXRpbmc-XG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMi4wLjAtcmMuMlwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjEuMC4wLXJjLjJcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wLXJjLjJcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjIuMC4wLXJjLjNcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0)
+
+You aren't able to use numbers directly, no. However, the foot gun of lacking clarity in your code that unearths makes preventing this input a pretty good idea. In a world where `px` is just one of many units that a CSS property could accept (`%`, `vh`, `vwmax`, `pt`, `em`, `rem`, `pt`, `pc`, ad (almost) infinitum) being explicit with your code with not only help your consumers, but it'll help your future self maintain the components that you create. On top of that, there are numbers that you'll eventually want to apply to your styles that won't beg for the application of a unit at all:
```

Comment:
"clarity in your code that unearths makes" -> "clarity in your code that doing so unearths makes"

Reply:
```suggestion
The current version of Lit's styleMap excludes numbers as values. You might expect Lit to automatically convert numbers to `px` values, but on second thought that isn't actually what you'd want. In CSS, where `px` is just one of many units that a numeric CSS property could accept (`%`, `vh`, `vwmax`, `pt`, `em`, `rem`, `pt`, `pc`, _ad infinitum_), there's no way for Lit to know or even assume what kind of number you're using. On top of that, you might want to apply unit-less numbers directly to your styles:
```